client/src/main/java/io/pravega/client/stream/impl/CheckpointState.java

```diff
@@ -256,6 +266,7 @@ private void write00(CheckpointState object, RevisionDataOutput output) throws I
             output.writeMap(object.uncheckpointedHosts, stringSerializer, (out, hosts) -> out.writeCollection(hosts, stringSerializer));
             output.writeMap(object.checkpointPositions, stringSerializer, (out, map) -> out.writeMap(map, segmentSerializer, longSerializer));
             output.writeMap(object.lastCheckpointPosition, segmentSerializer, longSerializer);
+            output.writeMap(object.lastStreamCutPosition, segmentSerializer, longSerializer);
```

Comment:
Please add tests to verify this change...

Reply:
+ Please add UTs and integration tests for the new code...

Reply:
Added two tests for the new code. Please have a look